### PR TITLE
Hibernate upgrade oscarehr common

### PR DESCRIPTION
**Pull request by sebastian-j-ibanez**
## Changes made
- Update queries to use positional parameters starting from 1
- Refactor query strings for readability
- Remove all string appending in favour of positional parameters
## Files changed
- Any `DaoImpl.java` file in `src/main/java/org/oscarehr/common/` that:
  - Starts with Oscar
  - Starts with the letter P-Z

## Summary by Sourcery

Upgrade Hibernate queries to use positional parameters starting from 1 across multiple DAO implementation files, enhancing readability and maintainability.

Enhancements:
- Refactor query strings to improve readability and maintainability by using positional parameters starting from 1.
- Remove string appending in favor of using positional parameters for query construction.

### CODE REVIEW

File: src/main/java/org/oscarehr/common/dao/PopulationReportDaoImpl.java [hunk 1]
```diff
@@ -183,9 +201,21 @@ public Map<Integer, Integer> getCaseManagementNoteCountGroupedByIssueGroup(int p
         ResultSet rs = null;
         try {
             c = DbConnectionFilter.getThreadLocalDbConnection();
-            String sqlCommand = "select issueGroupId,count(distinct casemgmt_note.note_id) from IssueGroupIssues,casemgmt_issue,casemgmt_issue_notes,casemgmt_note where IssueGroupIssues.issue_id=casemgmt_issue.issue_id and casemgmt_issue_notes.id=casemgmt_issue.id and casemgmt_note.note_id=casemgmt_issue_notes.note_id " + (encounterType == null ? "" : "and casemgmt_note.encounter_type=? ") + "and casemgmt_note.program_no=? " + (roleId == null ? "" : "and casemgmt_note.reporter_caisi_role=? ") + "and casemgmt_note.observation_date>=? and casemgmt_note.observation_date<=? group by issueGroupId";
-            ps = c.prepareStatement(sqlCommand);
             int counter = 1;
+            String sqlCommand = "select issueGroupId,count(distinct casemgmt_note.note_id) from IssueGroupIssues,casemgmt_issue,casemgmt_issue_notes," +
```
Comment (yingbull) [inline src/main/java/org/oscarehr/common/dao/PopulationReportDaoImpl.java:205]: A lot changed here, are we sure?

File: src/main/java/org/oscarehr/common/dao/PopulationReportDaoImpl.java [hunk 2]
```diff
@@ -212,13 +242,20 @@ public Map<Integer, Integer> getCaseManagementNoteCountGroupedByIssueGroup(int p
...
+            " and casemgmt_note.provider_no=?" + counter++ + " and casemgmt_note.observation_date>=?" + counter++ + 
+            " and casemgmt_note.observation_date<=?" + counter++ + " group by issueGroupId";
             ps = c.prepareStatement(sqlCommand);
-            ps.setString(1, encounterType.getOldDbValue());
-            ps.setInt(2, programId);
-            ps.setString(3, provider.getProviderNo());
-            ps.setTimestamp(4, new Timestamp(startDate != null ? startDate.getTime() : 0));
-            ps.setTimestamp(5, new Timestamp(endDate != null ? endDate.getTime() : System.currentTimeMillis()));
+            counter = 1;
+            ps.setString(counter++, encounterType.getOldDbValue());
```
Comment (yingbull) [inline src/main/java/org/oscarehr/common/dao/PopulationReportDaoImpl.java:254]: I squinted enough to follow.  This sort of logic even if it was sort of there might have been worth adding comments.

File: src/main/java/org/oscarehr/common/dao/ProviderLabRoutingDaoImpl.java [hunk 3]
```diff
@@ -44,12 +44,12 @@ public ProviderLabRoutingDaoImpl() {
 
     private List<ProviderLabRoutingModel> getProviderLabRoutings(Integer labNo, String labType, String providerNo,
                                                                  String status) {
-        Query q = entityManager.createQuery("select x from " + modelClass.getName()
-                + " x where x.labNo LIKE ? and x.labType LIKE ? and x.providerNo LIKE ? and x.status LIKE ?");
-        q.setParameter(0, labNo != null ? labNo : "%");
-        q.setParameter(1, labType != null ? labType : "%");
-        q.setParameter(2, providerNo != null ? providerNo : "%");
-        q.setParameter(3, status != null ? status : "%");
+        Query q = entityManager.createQuery("select x from ?1 x where x.labNo LIKE ?2 and x.labType LIKE ?3 and x.providerNo LIKE ?4 and x.status LIKE ?5");
```
[the remark this answers is not in the document]
Comment (yingbull) [reply src/main/java/org/oscarehr/common/dao/ProviderLabRoutingDaoImpl.java:47]: Agree, but these will need to be another optimization effort as we know these work so will leave as is during this upgrade.

File: src/main/java/org/oscarehr/common/dao/SecObjPrivilegeDaoImpl.java [hunk 4]
```diff
@@ -44,11 +44,11 @@ public SecObjPrivilegeDaoImpl() {
 
     @Override
     public List<SecObjPrivilege> findByRoleUserGroupAndObjectName(String roleUserGroup, String objectName) {
-        String sql = "select s FROM SecObjPrivilege s WHERE s.id.roleUserGroup = :rug AND  s.id.objectName = :obj";
+        String sql = "select s FROM SecObjPrivilege s WHERE s.id.roleUserGroup = ?1 AND  s.id.objectName = ?2";
 
         Query query = entityManager.createQuery(sql);
-        query.setParameter("obj", objectName);
-        query.setParameter("rug", roleUserGroup);
+        query.setParameter(1, objectName);
```
Comment (yingbull) [inline src/main/java/org/oscarehr/common/dao/SecObjPrivilegeDaoImpl.java:50]: Are these flipped?

File: src/main/java/org/oscarehr/common/dao/SiteDaoImpl.java [hunk 5]
```diff
@@ -62,24 +62,24 @@ public void save(Site s) {
...
+                query = entityManager.createNativeQuery("update scheduledate set reason = ?1 where reason = ?2");
+                query.setParameter(1, old.getName());
+                query.setParameter(2, s.getName());
                 query.executeUpdate();
 
-                query = entityManager.createNativeQuery("update appointment set location = :newname where location = :oldname");
-                query.setParameter("oldname", old.getName());
-                query.setParameter("newname", s.getName());
+                query = entityManager.createNativeQuery("update appointment set location = ?1 where location = ?2");
+                query.setParameter(1, old.getName());
```
Comment (yingbull) [inline src/main/java/org/oscarehr/common/dao/SiteDaoImpl.java:76]: Flipped?

File: src/main/java/org/oscarehr/common/dao/SiteDaoImpl.java [hunk 6]
```diff
@@ -62,24 +62,24 @@ public void save(Site s) {
...
+                query.setParameter(2, s.getName());
                 query.executeUpdate();
 
-                query = entityManager.createNativeQuery("update billing_on_cheader1 set clinic = :newname where clinic = :oldname");
-                query.setParameter("oldname", old.getName());
-                query.setParameter("newname", s.getName());
+                query = entityManager.createNativeQuery("update billing_on_cheader1 set clinic = ?1 where clinic = ?2");
+                query.setParameter(1, old.getName());
```
Comment (yingbull) [inline src/main/java/org/oscarehr/common/dao/SiteDaoImpl.java:81]: Flipped?

**Review by yingbull — CHANGES_REQUESTED** [review]

File: src/main/java/org/oscarehr/common/dao/OscarAppointmentDaoImpl.java [hunk 7]
```diff
@@ -723,16 +733,16 @@ public List<Object[]> findAppointmentsByDemographicIds(Set<String> demoIds, Date
                 "drugs.customName " +
```
[the remark this answers is not in the document]
Comment (sebastian-j-ibanez) [reply src/main/java/org/oscarehr/common/dao/OscarAppointmentDaoImpl.java:733]: It's something... that's for sure

File: src/main/java/org/oscarehr/common/dao/PopulationReportDaoImpl.java (hunk 2 above)
Comment (sebastian-j-ibanez) [reply src/main/java/org/oscarehr/common/dao/PopulationReportDaoImpl.java:254]: I can update these style of queries with comments.

Comment (sebastian-j-ibanez) [reply src/main/java/org/oscarehr/common/dao/PopulationReportDaoImpl.java]: They are now updated.

File: src/main/java/org/oscarehr/common/dao/SiteDaoImpl.java (hunk 5 above)
Comment (sebastian-j-ibanez) [reply src/main/java/org/oscarehr/common/dao/SiteDaoImpl.java]: Gave the SiteDaoImpl another look, I pushed a fix for this flip.

File: src/main/java/org/oscarehr/common/dao/SiteDaoImpl.java (hunk 6 above)
Comment (sebastian-j-ibanez) [reply src/main/java/org/oscarehr/common/dao/SiteDaoImpl.java]: Fix has been pushed.

File: src/main/java/org/oscarehr/common/dao/SecObjPrivilegeDaoImpl.java (hunk 4 above)
Comment (sebastian-j-ibanez) [reply src/main/java/org/oscarehr/common/dao/SecObjPrivilegeDaoImpl.java]: A fix has been pushed for this improper flip.

**Review by yingbull — APPROVED** [review]

**Pull request merged**